airbyte-workers/src/main/java/io/airbyte/workers/internal/EmptyAirbyteSource.java

```diff
@@ -83,16 +83,22 @@ public void start(final WorkerSourceConfig workerSourceConfig, final Path jobRoo
          */
         isResetBasedForConfig = false;
       } else {
-        stateWrapper = StateMessageHelper.getTypedState(workerSourceConfig.getState().getState(), useStreamCapableState);
+        if (workerSourceConfig.getState() != null ) {
+          stateWrapper = StateMessageHelper.getTypedState(workerSourceConfig.getState().getState(), useStreamCapableState);
 
-        if (stateWrapper.isPresent() &&
-            stateWrapper.get().getStateType() == StateType.LEGACY &&
-            !isResetAllStreamsInCatalog(workerSourceConfig)) {
-          log.error("The state a legacy one but we are trying to do a partial update, this is not supported.");
-          throw new IllegalStateException("Try to perform a partial reset on a legacy state");
+          if (stateWrapper.isPresent() &&
+              stateWrapper.get().getStateType() == StateType.LEGACY &&
+              !isResetAllStreamsInCatalog(workerSourceConfig)) {
+            log.error("The state a legacy one but we are trying to do a partial update, this is not supported.");
+            throw new IllegalStateException("Try to perform a partial reset on a legacy state");
+          }
+
+          isResetBasedForConfig = true;
+        } else {
+          /// No state
+          isResetBasedForConfig = false;
```

Comment:
Probably a good idea to add a unit test for this case, but I'm fine with us adding that later so that we can get this fix in ASAP